src/main/scala/higherkindness/rules_scala/workers/common/FileUtil.scala

```diff
@@ -62,6 +59,22 @@ object FileUtil {
     }
   }
 
+  private def lock[A](lockFile: Path)(f: => A):A = {
```

Comment:
It's pretty easy to have a race condition here. Not sure if the locking is the exact right path.

Reply:
Woops, just saw this comment. This lock seems safe to me, but maybe I'm missing something. What would the race condition look like?

Reply:
Let me add a comment to the code to illustrate